scheduler/preemption.go

```diff
@@ -435,6 +439,99 @@ OUTER:
 	return filteredBestAllocs
 }
 
+// allocDeviceGroup represents a group of allocs that share a device
+type allocDeviceGroup struct {
```

Comment:
I was thrown away by the name, as the struct doesn't describe device group of an alloc.  I would consider swapping the order, `deviceGroupAllocs`, or going more generic with `deviceGroupUsage`.  What do you think?